server/src/pages/shot/shotpage-header.js

```diff
@@ -91,7 +91,7 @@ exports.ShotPageHeader = class ShotPageHeader extends React.Component {
     if (this.props.isOwner) {
       return (
         <div className="shot-fxa-signin">
-          <SignInButton isFxaAuthenticated={this.props.isFxaAuthenticated} initialPage={this.props.shot.id}
+          <SignInButton isFxaAuthenticated={this.props.hasFxa} initialPage={this.props.shot.id}
```

Comment:
Is there any meaningful difference between `hasFxa` and `isFxaAuthenticated`? Do we need both?

Reply:
hasFxa determines its boolean value based off existence of accountId cookie, where as isFxaAuthenticated checks for both accountIdCookie and if the accountId cookie is linked to deviceId that has taken the shot ( const isFxaAuthenticated = req.accountId && req.accountId === req.shot.accountId;)

I have left isFxaAuthenticated in model as it's used by Favorite button enable/disable logic. I think now its safe to switch to hasFxa for favorite button logic as we are checking isOwner before displaying Favorite.

isOwner on shot page checks in database if shot is owned and linked to to a valid deviceId or accountId. I can update the PR with the change